bokeh/models/ranges.py

```diff
@@ -119,6 +142,16 @@ class FactorRange(Range):
     this categorical range.
     """)
 
+    bound_lower = Either(String, Int, help="""
+    The minimum value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
+
+    bound_upper = Either(String, Int, help="""
+    The max value that the range is allowed to go to - typically used to prevent
+    the user from panning/zooming/etc away from the data.
+    """)
```

Comment:
Well, I was thinking that we should have consistent naming across the ranges, as then once you've learned how to do it for one range, then you don't have to look up how to do it for a different one. Maybe we can pick something that works for all?

---
d3 defines a rangeExtent and scaleExtent that would be used in this context.

https://github.com/mbostock/d3/wiki/Ordinal-Scales#ordinal_rangeExtent
`ordinal.rangeExtent()`
Returns a two-element array representing the extent of the scale's range, i.e., the smallest and largest values.

https://github.com/mbostock/d3/wiki/zoom-behavior#scaleExtent
`zoom.scaleExtent([extent])`
Specifies the zoom scale's allowed range as a two-element array, [minimum, maximum]. If not specified, returns the current scale extent, which defaults to [0, Infinity].

---

I also had a poke around Grammar of Graphics and didn't see anything conclusive, but it's a big book so may have missed it.

---

ggplot2 - http://docs.ggplot2.org/current/xylim.html
has the concept of xlim and ylim
xlim(15, 20) 
for discrete data you can do: xlim("a", "b", "c")

---

vega has a concept of domainMin/Max and rangeMin/Max domainMin/Max is for quantitative scales only, but rangeMin/Max appears to be for all (I haven't seen an example with this in action though so not 100%)

https://github.com/vega/vega/wiki/Scales

---

plotly - couldn't find it

---

In general, looking around I'm not finding a clear message, but I may be looking in the wrong place. 

Maybe @pzwang has opinions about this?